generate_data.py

```diff
@@ -45,7 +45,15 @@ def main(args):
         dataset = DATASETS[args.dataset](dataset_root, args)
 
         if not args.iid:
-            if args.alpha > 0:  # Dirichlet(alpha)
+            if args.dataset == "domain":
+                with open(dataset_root / "original_partition.pkl", "rb") as f:
+                    partition = {}
+                    partition["data_indices"] = pickle.load(f)
+                    partition["separation"] = None
+                    args.client_num = len(partition["data_indices"])
+                with open(dataset_root / "original_stats.json", "r") as f:
+                    stats = json.load(f)
+            elif args.alpha > 0:  # Dirichlet(alpha)
```

Comment:
I moved `if args.dataset=='domain'` to the front of `elif args.alpha>0` to prevent code from executing `dirichlet()` when `--alpha >0`, since `dirichlet()` is not designed for DomainNet.